modules/ROOT/examples/live-demos/advtemplate-readonly-template/index.js

```diff
@@ -0,0 +1,347 @@
+    const resp = [...store.items]
+    return resp
+  }
+
+  const getCategory = (id) => {
+    validate(id, isValidId, `Invalid "id" parameter: ${id}`)
+    const category = categoryIndex[id]
+    if (typeof category !== 'undefined'){
+      return category
+    } else {
+      throw new StoreError(404, 'Category not found')
+    }
+  }
+
+  const getTemplate = (id) => {
+    validateId(id)
+    const res = templateIndex[id]
+    if (typeof res !== 'undefined'){
+      return res
+    } else {
+      throw new StoreError(404, 'Template not found')
+    }
+  }
+
+  const createCategory = (title, locked) => {
+    validateTitle(title)
+
+    const id = genId()
+    const category = {
+      id,
+      title,
+      locked,
+      items: []
+    }
+    categoryIndex[id] = category;
+    store.items = [...store.items, category]
+    return category
+  }
+
+  const createTemplate = (title, content, categoryId) => {
+    validateTitle(title)
+    validate(content, isNonEmptyString, `Invalid "content" parameter: ${content}`)
+    const id = genId()
+    const template = {
+      id,
+      title,
+      content,
+    }
+    const parent = getParent(categoryId)
+    parent.items = [...parent.items, template]
+    templateIndex[id] = [template, categoryId];
+    return template
+  }
+
+  const renameCategory = (id, title) => {
+    validateTitle(title)
+    const category = getCategory(id)
+    category.title = title;
+  }
+
+  const renameTemplate = (id, title) => {
+    validateTitle(title)
+    const [template] = getTemplate(id)
+    template.title = title;
+  }
+
+  const deleteTemplate = (id) => {
+    const [, categoryId] = getTemplate(id)
+    const parent = getParent(categoryId)
+    parent.items = filterOut(parent.items, id)
+    delete templateIndex[id]
+  }
+
+  const deleteCategory = (id) => {
+    const category = getCategory(id);
+    category.items.forEach((template) => deleteTemplate(template.id))
+    store.items = filterOut(store.items, id)
+    delete categoryIndex[category.id]
+  }
+
+  const moveTemplate = (id, newCategoryId) => {
+    const [template, categoryId] = getTemplate(id)
+    if (categoryId === newCategoryId) {
+      return
+    }
+    const oldParent = getParent(categoryId)
+    const newParent = getParent(newCategoryId)
+    newParent.items = [...newParent.items, template]
+    oldParent.items = filterOut(oldParent.items, id)
+    templateIndex[id] = [template, newCategoryId]
+  }
+
+  const moveCategoryItems = (id, newCategoryId) => {
+    const oldCategory = getCategory(id)
+    if (id === newCategoryId) {
+      return
+    }
+    oldCategory.items.forEach((item) => moveTemplate(item.id, newCategoryId))
+  }
+
+  const load = (items, id) => {
+    for (const item of items) {
+      if (item.items) {
+        const category = createCategory(item.title, item.locked);
+        load(item.items, category.id );
+      } else {
+        createTemplate(item.title, item.content, id);
+      }
+    }
+  };
```

Comment:
```suggestion
    items.forEach((item) => {
      if (item.items) {
        const category = createCategory(item.title, item.locked);
        load(item.items, category.id );
      } else {
        createTemplate(item.title, item.content, id);
      }
    });
```

Can be re-written as